lib/api/controllers/funnelController.js

```diff
@@ -443,6 +443,9 @@ class FunnelController {
         this.handleErrorDump(e);
         callback(e);
         return null;
+      })
+      .finally(() => {
+        delete this.pendingRequests[request.id];
```

Comment:
I would rather delete this request just after `getRequestSlot` answers us that we can process it, since it means that it isn't pending anymore. 

And since the overload protection mechanism can optionally be bypassed by plugins, this means I would add this [line 429](https://github.com/kuzzleio/kuzzle/blob/2381e29b0c68fa398321021287c8f0e68c3bc928/lib/api/controllers/funnelController.js#L429)

Reply:
Following our discussion, I just removed `pendingRequests`.
@ballinette: You may want to review again. We could not find any use for this queue, while it obviously induces some risks.